cmd/launcher/main.go

```diff
@@ -153,6 +143,63 @@ func runSubcommands() error {
 
 }
 
+// runNewerLauncherIfAvailable checks the autoupdate library for a newer version
+// of launcher than the currently-running one. If found, it will exec that version.
+func runNewerLauncherIfAvailable(ctx context.Context, logger log.Logger) {
+	newerBinary, err := latestLauncherPath(ctx, logger)
+	if err != nil {
+		logutil.Fatal(logger, err, "checking for updated version")
+	}
+
+	if newerBinary != "" {
+		level.Debug(logger).Log(
+			"msg", "preparing to exec new binary",
+			"oldVersion", version.Version().Version,
+			"newBinary", newerBinary,
+		)
+		if err := execwrapper.Exec(ctx, newerBinary, os.Args, os.Environ()); err != nil {
+			logutil.Fatal(logger, err, "exec")
+		}
+		panic("how")
+	} else {
+		level.Debug(logger).Log("msg", "Nothing newer")
+	}
+}
```

Comment:
```suggestion
		panic("how")
		return // linter might complain about this because not reachable, but we can just drop it then
	}
	
	// nothing new found
	level.Debug(logger).Log("msg", "Nothing newer")
}
```